### PR TITLE
Add stubs for new WordPress releases

### PR DESCRIPTION
**Pull request by swissspidy**
Adds 6.5 to `release-latest-versions.sh`

### CODE REVIEW

File: wordpress-stubs.php [hunk 1]
```diff
@@ -85557,6 +85583,18 @@ function admin_created_user_email($text)
     function wp_is_authorize_application_password_request_valid($request, $user)
     {
     }
+    /**
+     * Validates the redirect URL protocol scheme. The protocol can be anything except http and javascript.
+     *
+     * @since 6.3.2
```
Comment (szepeviktor) [inline wordpress-stubs.php]: There must be something wrong here ...

File: generate.sh [hunk 2]
```diff
@@ -52,7 +52,7 @@ if [ -r source/wordpress/wp-includes/Requests/Cookie/Jar.php ]; then
     )
     for METHOD in "${REQUESTS_V1_METHODS[@]}"; do
         # Get the line number where the method is defined.
-        LINE="$(php -r "require 'wordpress-stubs.php'; print (new ReflectionMethod('${METHOD}'))->getStartLine();")"
+        LINE="$(php -d error_reporting="E_ERROR" -r "require 'wordpress-stubs.php'; print (new ReflectionMethod('${METHOD}'))->getStartLine();")"
```
Comment (szepeviktor) [inline generate.sh:55]: releasing is peaceful on php 7.4

**Comment by szepeviktor** [comment]
not all WP releases are released here!

File: generate.sh (hunk 2 above)
Comment (swissspidy) [reply generate.sh:55]: Whoops, didn't mean to commit that. I suppose the script automatically stages all files and not just the stubs file.

FWIW I was using PHP 8.3 and got PHP warnings when loading the stubs file from older WordPress releases here.

Comment (szepeviktor) [reply generate.sh:55]: The WordPress ecosystem is not PHP 8 compatible.

**Comment by szepeviktor** [comment]
There must have been a security incident.

**Comment by swissspidy** [comment]
Yes there were a few security releases recently. For example 6.3.2  was released in October, see https://wordpress.org/documentation/wordpress-version/version-6-3-2/ and 6.3.3 in January, see https://wordpress.org/documentation/wordpress-version/version-6-3-3/

File: wordpress-stubs.php (hunk 1 above)
Comment (szepeviktor) [reply wordpress-stubs.php]: The commit says _Generate stubs for WordPress 6.2.3_

@swissspidy Please advise, you are The WordPress Superhero.

**Comment by szepeviktor** [comment]
It is not possible to push tags in a PR 😢

**Comment by swissspidy** [comment]
Yeah I was just about to say. I think you need to run the script on the main branch yourself because of all the tags. Or give me access 👿 😄

**Comment by IanDelMar** [comment]
The functionMap.php needs some changes.

File: wordpress-stubs.php [hunk 3]
```diff
@@ -4552,7 +4560,8 @@ protected function get_items_per_page($option, $default_value = 20)
          *
          * @since 3.1.0
          *
-         * @param string $which
+         * @param string $which The location of the pagination: Either 'top' or 'bottom'.
+         * @phpstan-param 'top'|'bottom' $which
          * @phpstan-param "top"|"bottom" $which
```
Comment (IanDelMar) [inline wordpress-stubs.php:4565]: Needs change in functionMap.php

File: wordpress-stubs.php [hunk 4]
```diff
@@ -4692,7 +4701,8 @@ protected function get_table_classes()
          * Generates the table navigation above or below the table
          *
          * @since 3.1.0
-         * @param string $which
+         * @param string $which The location of the navigation: Either 'top' or 'bottom'.
+         * @phpstan-param 'top'|'bottom' $which
          * @phpstan-param "top"|"bottom" $which
```
Comment (IanDelMar) [inline wordpress-stubs.php:4706]: Needs change in functionMap.php

File: wordpress-stubs.php [hunk 5]
```diff
@@ -6235,22 +6245,20 @@ public function rmdir($path, $recursive = \false)
...
+         *   owner: string|false,
+         *   group: string|false,
+         *   size: int|string|false,
+         *   lastmodunix: int|string|false,
+         *   lastmod: string|false,
+         *   time: string|false,
+         *   type: string,
+         *   files: array|false,
+         * }>
          * @phpstan-return false|array<string, array{name: string, perms: string, permsn: string, owner: string|false, size: int|string|false, lastmodunix: int|string|false, lastmod: string|false, time: string|false, type: 'f'|'d'|'l', group?: string|false, number?: int|string|false, files?: array|false}>
```
Comment (IanDelMar) [inline wordpress-stubs.php:6262]: Needs check

Comment (szepeviktor) [reply wordpress-stubs.php:6262]: I want to release stubs with bugs.

File: wordpress-stubs.php [hunk 6]
```diff
@@ -62545,32 +63877,202 @@ public function __construct()
          * @param array[][] $fonts Optional. The font-families and their font variations.
          *                         See {@see wp_print_font_faces()} for the supported fields.
          *                         Default empty array.
```
Comment (IanDelMar) [inline wordpress-stubs.php:63879]: Why has the @phpstan-param disappeared?

**Comment by herndlm** [comment]
what is going on here, why are there so many (old) commits :D

**Comment by swissspidy** [comment]
@herndlm Because there were many WordPress releases :)

**Comment by herndlm** [comment]
but we can only ever update the stubs and create a tag for one at a time AFAIK

**Comment by herndlm** [comment]
and going to an old one would basically revert stuff, which might look weird. I guess now I understand what Viktor meant with adding tags in a branch / PR hmm

File: wordpress-stubs.php (hunk 1 above)
Comment (szepeviktor) [reply wordpress-stubs.php]: @swissspidy Please says something. Is this a core error?

**Comment by szepeviktor** [comment]
Going to release everything up to 6.5.0 manually.
I leave core phpstan tag vs. our phpstan tag up to You guys.

**Pull request closed without merging**